policy/get-subtree-portfolio-v2.go

```diff
@@ -0,0 +1,173 @@
+	ProposalNumber int       `json:"proposalNumber" bigquery:"proposalNumber"`
+	NameDesc       string    `json:"nameDesc" bigquery:"nameDesc"`
+	Status         string    `json:"status" bigquery:"status"`
+	Contractor     string    `json:"contractor" bigquery:"contractor"`
+	Price          float64   `json:"price" bigquery:"price"`
+	PriceMonthly   float64   `json:"priceMonthly" bigquery:"priceMonthly"`
+	Producer       string    `json:"producer" bigquery:"producer"`
+	StartDate      time.Time `json:"startDate" bigquery:"startDate"`
+	EndDate        time.Time `json:"endDate" bigquery:"endDate"`
+	PaymentSplit   string    `json:"paymentSplit" bigquery:"paymentSplit"`
+}
+
+type GetSubtreePortfolioResp struct {
+	Policies []PolicyInfo `json:"policies"`
+}
+
+func GetSubtreePortfolioFx(w http.ResponseWriter, r *http.Request) (string, interface{}, error) {
+	var (
+		req  GetPoliciesReq
+		resp GetSubtreePortfolioResp
+	)
+
+	log.SetPrefix("[GetSubtreePortfolioFx] ")
+	defer log.SetPrefix("")
+	log.Println("Handler Start -----------------------------------------------")
+
+	idToken := r.Header.Get("Authorization")
+	authToken, err := models.GetAuthTokenFromIdToken(idToken)
+	if err != nil {
+		log.Printf("error getting authToken: %s", err.Error())
+		return "", nil, err
+	}
+
+	body := lib.ErrorByte(io.ReadAll(r.Body))
+	defer r.Body.Close()
+
+	log.Printf("request: %s", string(body))
+
+	err = json.Unmarshal(body, &req)
+	if err != nil {
+		log.Printf("error unmarshiling request: %s", err.Error())
+		return "", nil, err
+	}
+
+	producers := make([]string, 0)
+	if authToken.Role != models.UserRoleAdmin {
+		producers = append(producers, authToken.UserID)
+		children, err := getNodeChildren(authToken.UserID)
+		if err != nil {
+			log.Printf("error fetching node %s children: %s", authToken.UserID, err.Error())
+			return "", nil, err
+		}
+		producers = append(producers, children...)
+	}
+
+	resp.Policies, err = getPortfolioPoliciesV2(producers, req.Queries, req.Limit)
+	if err != nil {
+		log.Printf("error query: %s", err.Error())
+		return "", nil, err
+	}
+	log.Printf("found %02d policies", len(resp.Policies))
+
+	rawResp, err := json.Marshal(resp)
+
+	log.Printf("response: %s", string(rawResp))
+	log.Println("Handler end -------------------------------------------------")
+
+	return string(rawResp), resp, err
+}
+
+type Child struct {
+	Uid string `bigquery:"nodeUid"`
+}
+
+func getNodeChildren(nodeUid string) ([]string, error) {
+	baseQuery := fmt.Sprintf("SELECT nodeUid FROM `%s.%s` WHERE ", models.WoptaDataset, models.NetworkTreeStructureTable)
+	whereClause := fmt.Sprintf("rootUid = '%s'", nodeUid)
+	query := fmt.Sprintf("%s %s", baseQuery, whereClause)
+	result, err := lib.QueryRowsBigQuery[Child](query)
+	if err != nil {
+		log.Printf("error fetching children from BigQuery for node %s: %s", nodeUid, err.Error())
+		return nil, err
+	}
+
+	ch := make([]string, 0)
+	for _, r := range result {
+		ch = append(ch, r.Uid)
+	}
+
+	return ch, nil
+}
```

Comment:
can we have a single function? The same, or almost, is used in network.GetNodeSubTree line: 59-62